src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1020,6 +1023,30 @@ protected override void HandleInitializeRequestAsync(IRequestResponder<Initializ
             responder.SetResponse(initializeResponse);
         }
 
+        private bool m_canReverse = false;
+        private void UpdateCapabilities()
+        {
+            if (m_program !is IDebugReversibleEngineProgram160)
+                return;
+
+            bool canReverse = (m_program as IDebugReversibleEngineProgram160).CanReverse() == HRConstants.S_OK;
```

Comment:
>(m_program as IDebugReversibleEngineProgram160) [](http://example.com/codeflow?start=30&length=47)

nit: don't use `as` unless you are going to null check. I would probably suggest:

```
var revProgram = m_program as IDebugReversibleEngineProgram160;
if (revProgram == null)
    return;

...
```